### PR TITLE
Bug/24 password is exposed in explain virtual for exa connections

### LINKED ISSUE #24: Password is exposed in EXPLAIN VIRTUAL for EXA connections
If a connection is used in a Virtual Schema that stores username and password, this information should not be exposed in the resulting pushdown SQL. This works fine for JDBC connections, but not for the special EXA connection:
```
CREATE CONNECTION sys_connection TO 'jdbc:exa:localhost:42627' USER 'SYS' IDENTIFIED BY 'exasol';
CREATE VIRTUAL SCHEMA vs USING adapter.jdbc_adapter
WITH connection_name = 'SYS_CONNECTION' 
sql_dialect='EXASOL' schema_name = 'LS' 
import_from_exa='TRUE' exa_connection_string='localhost:42627';
```
Execution of `explain virtual select x from vs.t;` returns the pushdown:
 `IMPORT FROM EXA AT 'localhost:42627' USER 'SYS' IDENTIFIED BY 'exasol' STATEMENT 'SELECT * FROM "LS"."T"'`

**The question is why do we need a special `exa_connection_string`?** If this is redundant we can just use the information from the provided connection. However this would be changed behavior.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #24.

### CODE REVIEW

File: src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+        createVirtualSchemaWithTablesForJoinTest();
+        assertVsQuery("SELECT TL.C1, TL.C2, TR.C2 FROM TL LEFT JOIN TR ON TL.C1 = TR.C1 ORDER BY TL.C1", //
+                table() //
+                        .row("K1", "L1", "R1") //
+                        .row("K3", "L3", null) //
+                        .row(null, "L2", null) //
+                        .matches());
+    }
+
+    private void createVirtualSchemaWithTablesForJoinTest() {
```
Comment (chiaradiamarcelo) [inline src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java:480]: maybe we can move this method to line 468, right after the first place it's used

**Review by chiaradiamarcelo — COMMENTED** [review]
looks good! left some comments, most of them minor things

File: doc/dialects/exasol.md [hunk 2]
```diff
@@ -27,85 +27,101 @@ CREATE JAVA ADAPTER SCRIPT SCHEMA_FOR_VS_SCRIPT.ADAPTER_SCRIPT_EXASOL AS
...
-For the internally used `IMPORT FROM EXA` statement, the address from `EXA_CONNECTION_STRING` and the user name and password from the connection will be used.
 
-### Creating a Virtual Schema
+In this case you have to provide the additional `EXA_CONNECTION` which contains the name of the connection definition used for the internally used `IMPORT FROM EXA` command.
+
+That means that in this case you will have **two** named connections: a JDBC connection for the Virtual Schema adapter and an EXA connection for the EXALoader &mdash; which runs the `IMPORT`.
+
+Please refer to the [CREATE CONNECTION](https://docs.exasol.com/sql/create_connection.htm) documentation for more details about how to define an EXA connection.
+
+#### Creating a Virtual Schema With EXA Import
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply doc/dialects/exasol.md:52]: We are using AP-style headline capitalization throughout our projects:
> Capitalize Words With Four or More Letters (Associated Press style)

File: src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+                "SELECT NOW() - INTERVAL '1' MINUTE FROM " + getVirtualTableName(this.virtualSchema, table)));
+        assertThat(exception.getMessage(),
+                containsString("Attention! Using literals and constant expressions with datatype "
+                        + "`TIMESTAMP WITH LOCAL TIME ZONE` in Virtual Schemas can produce an incorrect results"));
+    }
+
+    /**
+     * The CastAssertionBuilder is a convenience class that helps formulating cast assertions in a more readable way.
+     */
+    static class CastAssertionBuilder {
```
Comment (chiaradiamarcelo) [inline src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java:534]: Just a nitpick, I would name this as `CastAssertionVerifier` but it's just a suggestion

File: src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+        private Table prepareInputTable() {
+            final Table table = this.parent.createSingleColumnTable(this.fromType);
+            for (final Object sourceValue : this.inputValues) {
+                table.insert(sourceValue);
+            }
+            return table;
+        }
+
+        private Builder prepareExpectation(final Object... expectedValues) {
+            this.acceptType = this.acceptType == null //
```
Comment (chiaradiamarcelo) [inline src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java:577]: Another nitpick, I personally prefer to use `Optional` in such cases to force the user to acknowledge and deal with cases where the `acceptType` might not be there.
Just a suggestion as well, so feel free to take it or leave it as it is.

File: src/test/java/com/exasol/adapter/dialects/exasol/AbstractRemoteExasolVirtualSchemaConnectionIT.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import com.exasol.dbbuilder.dialects.Table;
+
+/**
+ * This class contains the common parts of the variants of the Exasol Virtual Schema that connect to a remote Exasol
+ * instance or cluster.
+ */
+abstract class AbstractRemoteExasolVirtualSchemaConnectionIT extends AbstractExasolSqlDialectIT {
```
Comment (chiaradiamarcelo) [inline src/test/java/com/exasol/adapter/dialects/exasol/AbstractRemoteExasolVirtualSchemaConnectionIT.java:13]: Nitpick: As we have the `abstract` class modifier, do we also need it in the name? Can't we just name the class `RemoteExasolVirtualSchemaConnectionIT` ?

**Review by chiaradiamarcelo — APPROVED** [review]
Looks good! I just left some small suggestions. Feel free to take them or leave it as it is

**Pull request merged**